dask/dataframe/io/parquet/core.py

```diff
@@ -1045,18 +1051,14 @@ def to_parquet(
                 {"append": append, "compression": compression},
             )
         }
+
+        # Convert data_write + dsk to computable collection
+        graph = HighLevelGraph.from_collections(
+            final_name, dsk, dependencies=(data_write,)
+        )
+        out = Scalar(graph, final_name, "")
     else:
-        # NOTE: We still define a single task to tie everything together
-        # when we are not writing a _metadata file. We do not want to
-        # return `data_write` (or a `data_write.to_bag()`), because calling
-        # `compute()` on a multi-partition collection requires the overhead
-        # of trying to concatenate results on the client.
-        final_name = "store-" + data_write._name
-        dsk = {(final_name, 0): (lambda x: None, data_write.__dask_keys__())}
-
-    # Convert data_write + dsk to computable collection
-    graph = HighLevelGraph.from_collections(final_name, dsk, dependencies=(data_write,))
-    out = Scalar(graph, final_name, "")
+        out = data_write
 
     if compute:
         out = out.compute(**compute_kwargs)
```

Comment:
I'm not sure how I feel about the fact that `out` will now be a `Series` when `write_metadata_file` is `False`.  Maybe when `compute` is `True` we should do something like:

```python
    if compute:
        out.compute(**compute_kwargs)
        out = None
```

This would avoid the braking behavior of changing the return type of `to_parquet` for the default case of `compute=True`. However, we are still technically "breaking" the current API for any code that may be expecting
`to_parquet(..., compute=False)` to return `None.`

Reply:
I left a message after but for the moment I just reverted the default behaviour in case we use `compute=True`. The goal of this MR is to provide more advanced users some ways to release memory earlier.

I actually feel like this is something we could introduce in a tutorial (or in the DataFrame docs) on working with preemptible instances. Or perhaps in the [resilience](https://examples.dask.org/resilience.html) section